ykpack/src/types.rs

```diff
@@ -501,16 +501,53 @@ impl Display for BinOp {
     }
 }
 
+/// A debugging entry, mapping a DefId to its definition path string.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct SirDebug {
+    def_id: DefId,
+    def_path_str: String,
+}
+
+impl SirDebug {
+    pub fn new(def_id: DefId, def_path_str: String) -> Self {
+        Self {
+            def_id,
+            def_path_str,
```

Comment:
Why are we encoding the type (well, actually the wrong type since `str != String`!) as part of the variable name?

Reply:
It's the actual name of the concept from the Rust compiler ;)

https://github.com/rust-lang/rust/blob/299ef86e1f8b3e53154f834115752c719b611fa1/src/librustc/ty/print/pretty.rs#L1021-L1031

Reply:
Wouldn't it better be called just `def_path`? The `_str` bit is actively confusing IMHO.

Reply:
We can change it in our code, but I don't think it's worth our time to change it upstream.

Reply:
Don't worry, I was only thinking about this PR!